test/test_image.py

```diff
@@ -368,6 +368,16 @@ def test_decode_jpeg_cuda(mode, img_path, scripted):
     # Some difference expected between jpeg implementations
     assert (img.float() - img_nvjpeg.cpu().float()).abs().mean() < 2
 
+@needs_cuda
+def test_decode_image_cuda_raises():
+    data = torch.randint(0, 127, size=(255,), device="cuda", dtype=torch.uint8)
+    exception_raised = True
+    try:
+        decode_image(data)
+    except RuntimeError as e:
+        exception_raised = True
+    assert exception_raised
```

Comment:
It's best to use pytest for asserting exceptions:

```suggestion
    with pytest.raises(RuntimeError):
        decode_image(data)
```